test/test_db.py

```diff
@@ -129,105 +200,154 @@ def test_alias_nick(db):
+    db.delete_nick_value(nick, 'testkey')
+    assert db.get_nick_value(nick, 'testkey') is None, (
+        'Trying to delete a key must not create it.')
+
+    # the nick still exists, and its key as well
+    assert db.get_nick_value(nick, 'otherkey') == 'value', (
+        'This key must not be deleted by error.')
+
+
+@pytest.mark.parametrize('value', (
+    'string-value',
+    123456789,
+    'unicode-välûé',
+    ['structured', 'value'],
+))
+def test_get_nick_value(db: SopelDB, value):
+    nick = 'TestUser'
+    nick_id = db.get_nick_id(nick, create=True)
+
+    with db.session() as session:
+        nick_value = NickValues(
+            nick_id=nick_id,
+            key='testkey',
+            value=json.dumps(value, ensure_ascii=False),
+        )
+        session.add(nick_value)
+        session.commit()
+
+    assert db.get_nick_value(nick, 'testkey') == value
+    assert db.get_nick_value(nick, 'otherkey') is None
+    assert db.get_nick_value('NotTestUser', 'testkey') is None, (
+        'This key must be defined for TestUser only.')
+
+
+def test_get_nick_value_default(db: SopelDB):
+    assert db.get_nick_value("TestUser", "nokey") is None
+    assert db.get_nick_value("TestUser", "nokey", "default") == "default"
+
+
+def test_unalias_nick(db: SopelDB):
     nick = 'Embolalia'
     nick_id = 42
 
-    nn = Nicknames(nick_id=nick_id, slug=Identifier(nick).lower(), canonical=nick)
-    session.add(nn)
-    session.commit()
-
-    aliases = ['EmbölaliÅ', 'Embo`work', 'Embo']
-    for alias in aliases:
-        nn = Nicknames(nick_id=nick_id, slug=Identifier(alias).lower(), canonical=alias)
+    with db.session() as session:
+        nn = Nicknames(
+            nick_id=nick_id,
+            slug=Identifier(nick).lower(),
+            canonical=nick,
+        )
         session.add(nn)
         session.commit()
 
+    aliases = ['EmbölaliÅ', 'Embo`work', 'Embo']
+    with db.session() as session:
+        for alias in aliases:
+            nn = Nicknames(
+                nick_id=nick_id,
+                slug=Identifier(alias).lower(),
+                canonical=alias,
+            )
+            session.add(nn)
+            session.commit()
+
     for alias in aliases:
         db.unalias_nick(alias)
 
-    for alias in aliases:
-        found = session.query(Nicknames) \
-                       .filter(Nicknames.nick_id == nick_id) \
-                       .all()
-        assert len(found) == 1
+    with db.session() as session:
+        found = session.scalar(
+            select(func.count())
+            .select_from(Nicknames)
+            .where(Nicknames.nick_id == nick_id)
+        )
+        assert found == 1
 
+
+def test_unalias_nick_one_or_none(db: SopelDB):
+    # this will create the first version of the nick
+    db.get_nick_id('Exirel', create=True)
+
+    # assert you can't unalias a unique nick
     with pytest.raises(ValueError):
-        db.unalias_nick('Mister_Bradshaw')
+        db.unalias_nick('Exirel')
```

Comment:
Aw, can't we keep the Monty Python reference? 🥺

Reply:
Oh, I didn't get it, so I didn't left it here, given all other nicks are from Sopel dev or users. But I could put it back, sure.

Reply:
I kinda want to do more indirect references/jokes like your lists of cheeses, rather than people's nicks. So when I write new tests (I think I wrote this one?) I try to insert those instead of using dev/user names.

Reply:
Oh, I see! Then I can change them all in one commit, that would make sense.

Reply:
Done.